### PR TITLE
Improve get object performance in interpreter

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/interpreter/ByteCode.h [hunk 1]
```diff
@@ -1144,20 +1154,24 @@ struct GetObjectInlineCacheData {
...
+    ObjectStructurePropertyName m_propertyName;
 };
 
 struct GetObjectInlineCacheSimpleCase : public gc {
-    GetObjectInlineCacheSimpleCase()
+    GetObjectInlineCacheSimpleCase(ObjectStructurePropertyName propertyName)
+        : m_propertyName(propertyName)
     {
-        memset(this, 0, sizeof(GetObjectInlineCacheSimpleCase));
+        memset(m_cachedStructures, 0, sizeof(ObjectStructure*) * inlineBufferSize);
```
Comment (clover2123) [inline src/interpreter/ByteCode.h:1174]: This change initializes only the `m_cachedStructures` with 0, not including the `m_cachedIndexes` array.
Initialization of `m_cachedIndexes` is not necessary?

File: src/interpreter/ByteCode.h [hunk 2]
```diff
@@ -1205,11 +1220,11 @@ class GetObjectPreComputedCase : public ByteCode {
     union {
         GetObjectInlineCacheSimpleCase* m_simpleInlineCache;
         GetObjectInlineCacheComplexCase* m_complexInlineCache;
```
Comment (clover2123) [inline src/interpreter/ByteCode.h:1222]: Just a suggestion about naming.
The name of cache structures `GetObjectInlineCacheSimpleCase` and `GetObjectInlineCacheComplexCase` is somewhat confusing with the bytecode `GetObjectPreComputedCase`.
How about renaming these structures by adding `Data` suffix similar to the cache structure in property setting (`SetObjectInlineCacheData`)?

File: src/interpreter/ByteCode.h (hunk 1 above)
Comment (ksh8281) [reply src/interpreter/ByteCode.h:1174]: Yes. we should init m_cachedStructures only

File: src/interpreter/ByteCode.h (hunk 2 above)
Comment (ksh8281) [reply src/interpreter/ByteCode.h:1222]: That's a good idea!

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**